client/core/types.go

```diff
@@ -527,54 +531,42 @@ type dexAccount struct {
 func newDEXAccount(acctInfo *db.AccountInfo) *dexAccount {
 	return &dexAccount{
 		host:      acctInfo.Host,
-		encKey:    acctInfo.EncKey,
+		encKey:    acctInfo.EncKey(),
 		dexPubKey: acctInfo.DEXPubKey,
 		isPaid:    acctInfo.Paid,
 		feeCoin:   acctInfo.FeeCoin,
 		cert:      acctInfo.Cert,
 	}
 }
 
-func (a *dexAccount) dbInfo() *db.AccountInfo {
-	return &db.AccountInfo{
-		Host:      a.host,
-		Cert:      a.cert,
-		EncKey:    a.encKey,
-		DEXPubKey: a.dexPubKey,
-		FeeCoin:   a.feeCoin,
-	}
-}
-
 // ID returns the account ID.
 func (a *dexAccount) ID() account.AccountID {
 	a.keyMtx.RLock()
 	defer a.keyMtx.RUnlock()
 	return a.id
 }
 
-// setupEncryption creates and returns a new privkey for the account.
+// setupCryptoLegacy creates and returns a new privkey for the account.
 // Returns existing privkey if previously set up.
-func (a *dexAccount) setupEncryption(crypter encrypt.Crypter) (*secp256k1.PrivateKey, error) {
+func (a *dexAccount) setupCryptoLegacy(crypter encrypt.Crypter) (encPriv, pubB []byte, err error) {
 	// Check if privKey exists. Check a.encKey instead of a.privKey
 	// as a.privKey will be nil if the account is locked.
 	a.keyMtx.RLock()
-	if a.encKey != nil {
-		defer a.keyMtx.RUnlock()
-		// no need to generate a new privKey, return existing privKey AFTER unlocking.
-		err := a.unlock(crypter)
-		return a.privKey, err
-	}
+	encKey := a.encKey
 	a.keyMtx.RUnlock()
+	if encKey != nil {
+		return nil, nil, fmt.Errorf("key already set")
+	}
```

Comment:
This is a change in behavior, but I couldn't figure out why this shouldn't be an error. The only place we are using this function is in `Register`, where the presence of a encrypted key would signal an unexpected state.